components/skin/ui/src/main/resources/PhenoTips/Skin.xml

```diff
@@ -560,16 +560,30 @@ document.observe('xwiki:suggest:containerCreated', function(event) {
         var recordFormData = function() {
           _this.formData = _this.form.serialize();
         };
+        var bottomButtons = $$("input[name^='action_']");
+        var updateSaveState = function(saving, recordData) {
+          // Record the form's current save state.
+          _this.form._saving = saving;
+          // Buttons should be disabled when the form is saving, and enabled when not.
+          bottomButtons.each(function(button) {
+            saving &amp;&amp; button.addClassName('disabled') || button.removeClassName('disabled');
+          });
+          recordData &amp;&amp; recordFormData();
+        };
         // Some form elements are still being modified while "xwiki:dom:loaded" is fired.
         // If the initial state of the form is recorded immediately, there will likely be
         // differences with the state of the form when navigating away even if the user
         // hasn't actually changed any data.
         // Setting a timeout, even zero, delays the initialization until other
         // form elements have been modified already.
         setTimeout(recordFormData, 0);
+        document.observe('xwiki:actions:save', updateSaveState.bind(this, true, false));
         // Record the state of the form again once the form is saved.
-        document.observe('xwiki:document:saved', recordFormData);
-        $$("input[name^='action_']").invoke('observe', 'click', recordFormData);
+        document.observe('xwiki:document:saved', updateSaveState.bind(this, false, true));
+        // Make sure to unlock form saving if save failed.
+        document.observe("xwiki:document:saveFailed", updateSaveState.bind(this, false, false));
+        document.observe("phenotips:scrollValidation:invalid", updateSaveState.bind(this, false, false));
```

Comment:
I don't like this dependency. Ideally, the `xwiki:actions:save` event shouldn't be sent at all, but the order of the events is reversed: first the event is sent, then the validation script stops the form submission. So, it seems that there is no other way at the moment to detect that the validation failed and the form submission was stopped.

The whole validation script should be refactored, starting with the upstream `LiveValidation`, but at the moment this is good enough.